VirtoCommerce.Platform.Web/Scripts/app/security/security.js

```diff
@@ -115,18 +115,33 @@
                 url: '/changepassword',
                 templateUrl: '$(Platform)/Scripts/app/security/dialogs/changePasswordDialog.tpl.html',
                 params: {
-                    userName: null,
                     onClose: null
                 },
-                controller: ['$scope', '$stateParams', 'platformWebApp.accounts', '$state', 'platformWebApp.authService', function ($scope, $stateParams, accounts, $state, authService) {
-                    $scope.userName = $stateParams.userName;
+                controller: ['$q', '$scope', '$stateParams', 'platformWebApp.accounts', 'platformWebApp.authService', 'platformWebApp.passwordValidationService', function ($q, $scope, $stateParams, accounts, authService, passwordValidationService) {
+                    $scope.userName = authService.userName;
+                    $scope.passwordValidationResult = {
+                        passwordIsValid: true,
+                        minPasswordLength: 0,
+                        errors: []
+                    };
 
                     accounts.get({ id: $stateParams.userName }, function (user) {
                         if (!user || !user.passwordExpired) {
                             $stateParams.onClose();
                         }
                     });
 
+                    $scope.validatePasswordAsync = function(value) {
+                        var promise = passwordValidationService.validatePasswordAsync(value).then(
+                            function (result) {
+                                $scope.passwordValidationResult = result;
+
+                                return result.passwordIsValid ? $q.resolve() : $q.reject();
```

Comment:
Why do you use `$q.resolve() : $q.reject();` this? I think all can work without these calls. Just returns result.passwordIsValid  should be enought.

Reply:
I did it because of `ui-validate-async` nature - it [expects](https://github.com/angular-ui/ui-validate/blob/master/src/validate.js#L108) to receive one of the following objects:
* Promise - this way control will be valid if the promise was resolved, and invalid if the promise was rejected;
* Anything else - this way `uiValidateAsync` directive will wrap it to promise and resolve/reject it if the value can or can't be implicitly converted to `true`.

If we'll replace `return result.passwordIsValid ? $q.resolve() : $q.reject()` with just `return result.passwordIsValid`, `validatePasswordAsync()` method will always return resolved promise with `passwordIsValid` as its result, because the HTTP call to password validation method always returns `Ok()`, even if password is invalid. So the password input will always stay valid, because the `ui-validate-async` directive looks only for resolution/rejection of the promise and ignores resolution result.

Reply:
If this  function returns `result.passwordIsvalid` boolean value which  will normally be handled by `ui-validate-async`  directive in else statement  https://github.com/angular-ui/ui-validate/blob/master/src/validate.js#L111

Reply:
I tried that, and the password input always stays valid. Apparently, it happens because `validatePasswordAsync()` still returns promise, so it gets handled by `if` statement on [line 108](https://github.com/angular-ui/ui-validate/blob/master/src/validate.js#L108).